### DISCUSSION TITLE
MacOS windowid

### DISCUSSION THREAD

**Discussion by chazhenry** [airsdk/Adobe-Runtime-Support#2580] [General]
My captive runtime app quietly runs the screencapture command on MacOS and records a short video. I'd like to record just my window (or a portion thereof), but I've never been able to get my windows windowid or coordinates. If I could, I could use:
`screencapture -l <windowid> `
or 
`screencapture -R x,y,w,h`

Is there an actionscript way to get the windowID?

**Comment by ajwfrost** [comment]
Can you not get the window coordinates with [NativeWindow.bounds](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display/NativeWindow.html#bounds)?

Otherwise you may be able to create an ANE for this, you should be able to get the NSWindow handle via `FREAcquireNativeWindowHandle`?

thanks

**Comment by chazhenry** [comment]
I can get NativeWindow.bounds, but it gives nonsense numbers. For my 1800x900 window centered on an HD monitor, I get:  
location = 760,426
size = 960,540

**Comment by chazhenry** [comment]
Hang on... Maybe I was getting the bounds too early. Moved it to follow my addChild statements. While they don't affect the size, I'm getting real numbers now. Thanks!

**Reply by ajwfrost** [reply]
That's good! I thought that sort of approach should have worked.. there may be some discrepancies in numbers because of the size of the window chrome stuff...

**Comment by chazhenry** [comment]
Spoke too soon. I'm getting good numbers, but I can't pass the needed string to the NativeProcess. The switch needs to read:
`-R70,118,1900,800   // -Rx,y,w,h`
But whether you double quote the arg or not, the commas are interpreted as an addition arg.  I've tried giving an array and also pushing the args onto a string vector. No joy. The NativeProcess always breaks the args at the commas.

```
   var nativeProcessStartupInfo:NativeProcessStartupInfo = new NativeProcessStartupInfo();
   var processArgs:Vector.<String> = new Vector.<String>();
   processArgs.push("-R70,118,1900,800");  // tried this
   processArgs.push("\"-R70,118,1900,800\"");    // and this
   nativeProcessStartupInfo.arguments = processArgs;
   process = new NativeProcess();
   process.start(nativeProcessStartupInfo);
```
